### DISCUSSION TITLE
confusing behaviour of the unit parameter of Param

### DISCUSSION THREAD

**Discussion by Real-John-Cheung** [abey79/vsketch#447] [Q&A]
When I use the `unit` parameter in `vsketch.Param` the value seems to be multiplied by something (I assume that convert the value to CSS pixel?) However `vsk.scale()` will completely ignore that which makes it very confusing.

For example, for the following code
```py
import vsketch


class BugSketch(vsketch.SketchClass):
    # Sketch parameters:
    a = vsketch.Param(1.0, unit = "cm")
    b = vsketch.Param(1, unit = "cm")
    def draw(self, vsk: vsketch.Vsketch) -> None:
        vsk.size("a4", landscape=False)
        vsk.scale("cm")
        print(self.a, self.b)
        vsk.rect(0, 0, self.a, self.b)

    def finalize(self, vsk: vsketch.Vsketch) -> None:
        vsk.vpype("linemerge linesimplify reloop linesort")


if __name__ == "__main__":
    BugSketch.display()
```

`self.a` and `self.b` will be something around 37, so I get a rectangle that's larger than the paper instead of a square of 1cm by 1cm. I am not sure if that's the desired behaviour it should have but for me I will say that's quite counter intuitive

**Comment by abey79** [comment]
Yeah, `unit` and `vsk.scale()` dont play well together as they have a "multiplicative" interaction.

My preferred workflow is avoid `vsk.scale()` and use `vsketch.Param(..., unit=...)`. That way, everything is in CSS pixel (the default vsketch unit), but the UI for parameters does the $UNIT <-> CSS pixel conversion. In that case, I would use `vsk.scale()` only for relative scaling, if at all.

The alternative workflow is also possible when starting with `vsk.scale("cm")`. This means that "_all_ units are now cm instead of CSS pixels". In that case, `unit=...` should really be avoided, because you dont want the UI to apply further scaling.
